### PR TITLE
AST DOM tests for EitherOrMultiPattern nodes

### PR DESCRIPTION
**Pull request by subyssurendran666**
## What it does

The PR contains the AST DOM tests for EitherOrMultiPattern nodes.
https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2087

## Author checklist

- [x] I have thoroughly tested my changes
- [x] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [x] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

File: org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverterEitherOrMultiPatternTest.java [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+		node = getASTNode(compilationUnit, 0, 0, 0);
+		assertEquals("variable decleration statement", node.getNodeType(), ASTNode.VARIABLE_DECLARATION_STATEMENT);
+
+		node = getASTNode(compilationUnit, 0, 0, 1);
+		assertEquals("variable decleration statement", node.getNodeType(), ASTNode.SWITCH_STATEMENT);
+
+		List<ASTNode> blockStatements= block.statements();
+		assertTrue("Number of statements not 2", blockStatements.size() == 2);
+
+		SwitchStatement switchDeclerationStatement = (SwitchStatement)blockStatements.get(1);
```
Comment (mpalat) [inline org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverterEitherOrMultiPatternTest.java:129]: nitpicking: spelling :)

File: org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverterEitherOrMultiPatternTest.java [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+		if (!isJRE22) {
+			printJREError();
+			return;
+		}
+		String contents = """
+					public class X {
+					  public static void main(String[] args) {
+					    Object notification = "Email notification";
+					    switch (notification) {
+					      case PushNotification _, EmailNotification _, SMSNotification(String _) when notification instanceof Object :
```
Comment (mpalat) [inline org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverterEitherOrMultiPatternTest.java:83]: Good that record pattern also added.

**Review by mpalat — CHANGES_REQUESTED** [review]

File: org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverterEitherOrMultiPatternTest.java [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+
+	public void setUpSuite() throws Exception {
+		super.setUpSuite();
+		this.ast = AST.newAST(getASTLatest(), true);
+		this.currentProject = getJavaProject("Converter_22");
+		if (this.ast.apiLevel() == AST.JLS22) {
+			this.currentProject.setOption(JavaCore.COMPILER_COMPLIANCE, JavaCore.VERSION_22);
+			this.currentProject.setOption(JavaCore.COMPILER_SOURCE, JavaCore.VERSION_22);
+			this.currentProject.setOption(JavaCore.COMPILER_CODEGEN_TARGET_PLATFORM, JavaCore.VERSION_22);
+			this.currentProject.setOption(JavaCore.COMPILER_PB_ENABLE_PREVIEW_FEATURES, JavaCore.ENABLED);
```
[the remark this answers is not in the document]
Comment (subyssurendran666) [reply org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverterEitherOrMultiPatternTest.java:52]: JEP 456 is currently not a standard feature.

File: org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverterEitherOrMultiPatternTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (subyssurendran666) [reply org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverterEitherOrMultiPatternTest.java:90]: Done

File: org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverterEitherOrMultiPatternTest.java (hunk 1 above)
Comment (subyssurendran666) [reply org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverterEitherOrMultiPatternTest.java]: Done. Corrected the spelling

File: org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverterEitherOrMultiPatternTest.java (hunk 2 above)
Comment (subyssurendran666) [reply org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverterEitherOrMultiPatternTest.java:90]: Thanks

File: org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverterEitherOrMultiPatternTest.java [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+					}
+					class EmailNotification {
+						String eMessage;
+						public EmailNotification(String eMessage) {
+							this.eMessage = eMessage;
+						}
+					}
+					record SMSNotification(Object smsMessage) {}
+				""";
+		this.workingCopy = getWorkingCopy("/Converter_22/src/X.java", false/*resolve*/);
```
[the remark this answers is not in the document]
Comment (subyssurendran666) [reply org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverterEitherOrMultiPatternTest.java]: Done

**Comment by mpalat** [comment]
@subyssurendran666 please synch with master in the light of https://github.com/eclipse-jdt/eclipse.jdt.core/pull/2282

**Pull request closed without merging**